File: 2nd_quarter/week06/Whaeun/숫자 짝꿍.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

string solution(string X, string Y) {
    string answer = "";
    
    
    int idx1 = 0, idx2 = 0;
    
    sort(X.begin(), X.end());
    sort(Y.begin(), Y.end());
    
    while(idx1 < X.size() && idx2 < Y.size()){
        if(X[idx1] == Y[idx2]){
            answer += X[idx1];
            idx1++;
            idx2++;
        }
        else if(X[idx1] > Y[idx2]){
            idx2++;
        }
        else{
            idx1++;
        }
        
    }
    
    if(answer == ""){
        return "-1";
    }
    
    sort(answer.begin(), answer.end(), greater<>());
    
    if(answer[0] == '0'){
        answer = "0";
    }
    
    return answer;
}

File: 2nd_quarter/week06/Royce/숫자 짝꿍.py
def solution(X, Y):
    answer = []
    
    for i in (set(X) & set(Y)):
        for j in range(min(X.count(i), Y.count(i))):
            answer.append(i)
    # X에서의 i개수와 Y에서의 i개수 중 작은 값에 대한 횟수 만큼 answer에 i를 넣어준다.
    
    answer.sort(reverse=True)
    
    # 아예 없을 경우에 -1을 리턴한다.
    if len(answer) == 0:
        return "-1"
    if answer[0] == "0":
        return "0"
    # 정렬을 했는데도 제일 앞에 있는게 0일 경우 그냥 0을 리턴한다.
    
    answer = "".join(answer)
    
    
    
    return answer
